src/algorithms.ts

```diff
@@ -50,3 +51,38 @@ export function intersectTokenScores(arrays: TokenScore[][]): TokenScore[] {
 
   return result;
 }
+
+export function prioritizeTokenScores(arrays: TokenScore[][]): TokenScore[] {
+  const tokenMap: Record<string, number> = {};
+
+  const mapsLength = arrays.length;
+  for (let i = 0; i < mapsLength; i++) {
+    const arr = arrays[i];
+
+    const entriesLength = arr.length;
+    for (let j = 0; j < entriesLength; j++) {
+      const [token, score] = arr[j];
+
+      if (token in tokenMap) {
+        tokenMap[token] += score + 0.5
+      } else {
+        tokenMap[token] = score;
+      }
+    }
+  }
+
+  return Object.entries(tokenMap).sort((a, b) => b[1] - a[1]);
```

Comment:
@ShogunPanda what if we adopt sparse array insertion here instead of sorting the resulting array?

Reply:
We could, but then you have to filter the array anyway.

Reply:
Ok let's re-iterate on this in the future